docs/integrations/sources/s3.md

```diff
@@ -237,11 +237,29 @@ The Avro parser uses the [Fastavro library](https://fastavro.readthedocs.io/en/l
 
 There are currently no options for JSONL parsing.
 
+### Markdown/PDF/Docx Format (Experimental)
+
+:::warning
+The Markdown/PDF/Docx format is currently an experimental feature and not subject to SLAs. Use at your own risk.
+:::
+
+The Markdown/PDF/Docx format is a special format that allows you to extract text from Markdown, PDF, and Word documents. If selected, the connector will extract text from the documents and output it as a single column named `content`. The content of the document will contain markdown formatting converted from the original file format.
+
+One record will be emitted for each document. If the content of the document is longer than 4MB, the document will be split up into multiple records, each with a maximum size of 4MB. The fields "chunk_number" (starting at 0) and "no_of_chunks" will be added to each record to indicate the order of the chunks and the total number of chunks. The field "id" will be added to each record containing the file name and chunk id and is suitable as a primary key.
+
+If a file with another format is encountered, it is skipped. The sync will not be stopped in this case.
```

Comment:
I think I agree that if a file is unreadable, it should probably stop the sync. This then delegates the exclusion of non-qualifying types to a glob pattern - or else the selection of a file type also more tightly constrains the glob pattern to just qualifying file extensions.

For example:

- If we have format set to PDF and there's a CSV incorrectly suffixed with `.pdf`, this will fail. Because our file-type-aware glob pattern (`/path/to/myfolder/**/*.pdf`) picked up a file that was unreadable.
- If we have format set to PDF and there's a CSV in the same directory, the CSV will be skipped. Because our file-type-aware glob pattern (`/path/to/myfolder/**/*.pdf`) was not concerned with files suffixed by `.csv`.

I'm using "glob pattern" as one concept, but how we implement may actually be a combination of other processes and patterns, for loops, etc.